CI: unit test TestMemKVClient  failed, panic: send on closed channel

**What happened**:

https://github.com/juicedata/juicefs/actions/runs/13690160180/job/38281823439
```
=== RUN   TestMemKVClient
2025/03/06 03:04:19.897224 juicefs[17096] <INFO>: Create session 1 OK with version: 1.3.0-dev+unknown [NewSession@base.go:519]
2025/03/06 03:04:19.998957 juicefs[17096] <INFO>: flush session 1: [FlushSession@base.go:700]
2025/03/06 03:04:19.999017 juicefs[17096] <INFO>: close session 1: <nil> [CloseSession@base.go:689]
2025/03/06 03:04:20.099898 juicefs[17096] <WARNING>: File name is too long as a trash entry, truncating it: fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff -> 1-24-ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff [trashEntry@base.go:2445]
2025/03/06 03:04:20.099939 juicefs[17096] <WARNING>: File name is too long as a trash entry, truncating it: fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff -> 1-24-ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff [trashEntry@base.go:2445]
2025/03/06 03:04:20.100362 juicefs[17096] <INFO>: cleanup trash: deleted 10 files in 140.223µs [CleanupTrashBefore@base.go:2499]
2025/03/06 03:04:20.452414 juicefs[17096] <INFO>: Quota for inode 15 is deleted [loadQuotas@quota.go:288]
2025/03/06 03:04:20.677949 juicefs[17096] <INFO>: flush session 1: [FlushSession@base.go:700]
2025/03/06 03:04:20.678243 juicefs[17096] <INFO>: close session 1: <nil> [CloseSession@base.go:689]
2025/03/06 03:04:21.904814 juicefs[17096] <INFO>: flush session 1: [FlushSession@base.go:700]
2025/03/06 03:04:21.904880 juicefs[17096] <INFO>: close session 1: <nil> [CloseSession@base.go:689]
2025/03/06 03:04:21.905429 juicefs[17096] <INFO>: Create session 2 OK with version: 1.3.0-dev+unknown [NewSession@base.go:519]
2025/03/06 03:04:21.916324 juicefs[17096] <INFO>: flush session 2: [FlushSession@base.go:700]
2025/03/06 03:04:21.916410 juicefs[17096] <INFO>: close session 2: <nil> [CloseSession@base.go:689]
2025/03/06 03:04:21.924522 juicefs[17096] <INFO>: Update session 2 OK with version: 1.3.0-dev+unknown [NewSession@base.go:519]
2025/03/06 03:04:26.936816 juicefs[17096] <INFO>: Quota for inode 4474 is deleted [loadQuotas@quota.go:288]
2025/03/06 03:04:26.936864 juicefs[17096] <INFO>: flush session 2: [FlushSession@base.go:700]
2025/03/06 03:04:26.938912 juicefs[17096] <INFO>: close session 2: <nil> [CloseSession@base.go:689]
2025/03/06 03:04:32.968678 juicefs[17096] <WARNING>: nlink of /check should be 3, but got 0 [Check@base.go:2025]
2025/03/06 03:04:32.968721 juicefs[17096] <WARNING>: Path /check (inode 4497) can be repaired, please re-run with '--path /check --repair' to fix it [Check@base.go:2052]
2025/03/06 03:04:32.968802 juicefs[17096] <WARNING>: nlink of /check should be 3, but got 0 [Check@base.go:2025]
2025/03/06 03:04:32.968877 juicefs[17096] <WARNING>: nlink of /check/d1/d2 should be 3, but got 0 [Check@base.go:2025]
2025/03/06 03:04:32.972313 juicefs[17096] <WARNING>: nlink of / should be 11, but got 3 [Check@base.go:2025]
2025/03/06 03:04:32.979636 juicefs[17096] <WARNING>: nlink of /check/d1 should be 3, but got 0 [Check@base.go:2025]
2025/03/06 03:04:32.979703 juicefs[17096] <WARNING>: nlink of /check/d1/d2/d3 should be 3, but got 0 [Check@base.go:2025]
2025/03/06 03:04:32.979736 juicefs[17096] <WARNING>: nlink of /check/d1/d2/d3/d4 should be 2, but got 0 [Check@base.go:2025]
2025/03/06 03:04:32.991640 juicefs[17096] <INFO>: Update session 2 OK with version: 1.3.0-dev+unknown [NewSession@base.go:519]
2025/03/06 03:04:37.498614 juicefs[17096] <INFO>: flush session 2: [FlushSession@base.go:700]
panic: send on closed channel

goroutine 1643131 [running]:
github.com/juicedata/juicefs/pkg/meta.(*baseMeta).deleteSlice(0xc020828f08, 0x1, 0x100000)
	/home/runner/work/juicefs/juicefs/pkg/meta/base.go:2391 +0x118
github.com/juicedata/juicefs/pkg/meta.(*kvMeta).deleteChunk(0xc022c43d80, 0x1197, 0x0)
	/home/runner/work/juicefs/juicefs/pkg/meta/tkv.go:2263 +0x35a
github.com/juicedata/juicefs/pkg/meta.(*kvMeta).doDeleteFileData(0xc022c43d80, 0x1197, 0x1001)
	/home/runner/work/juicefs/juicefs/pkg/meta/tkv.go:2287 +0x286
github.com/juicedata/juicefs/pkg/meta.(*baseMeta).tryDeleteFileData.func1()
	/home/runner/work/juicefs/juicefs/pkg/meta/base.go:2371 +0x3f
created by github.com/juicedata/juicefs/pkg/meta.(*baseMeta).tryDeleteFileData in goroutine 1621877
	/home/runner/work/juicefs/juicefs/pkg/meta/base.go:2370 +0xf5
FAIL	github.com/juicedata/juicefs/pkg/meta	237.308s
FAIL
make: *** [Makefile:90: test.meta.core] Error 1
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

--- a/pkg/meta/base.go
+++ b/pkg/meta/base.go
@@ -538,6 +538,8 @@ func (m *baseMeta) NewSession(record bool) error {
 }
 
 func (m *baseMeta) startDeleteSliceTasks() {
+	m.Lock()
+	defer m.Unlock()
 	if m.conf.MaxDeletes <= 0 || m.dslices != nil {
 		return
 	}
@@ -545,31 +547,35 @@ func (m *baseMeta) startDeleteSliceTasks() {
 	m.dSliceWG.Add(m.conf.MaxDeletes)
 	m.dslices = make(chan Slice, m.conf.MaxDeletes*10240)
 	for i := 0; i < m.conf.MaxDeletes; i++ {
-		go func() {
+		go func(dslices chan Slice) {
 			defer m.sessWG.Done()
 			defer m.dSliceWG.Done()
 			for {
 				select {
 				case <-m.sessCtx.Done():
 					return
-				case s, ok := <-m.dslices:
+				case s, ok := <-dslices:
 					if !ok {
 						return
 					}
 					m.deleteSlice_(s.Id, s.Size)
 				}
 			}
-		}()
+		}(m.dslices)
 	}
 }
 
 func (m *baseMeta) stopDeleteSliceTasks() {
+	m.Lock()
 	if m.conf.MaxDeletes <= 0 || m.dslices == nil {
+		m.Unlock()
 		return
 	}
 	close(m.dslices)
-	m.dSliceWG.Wait()
 	m.dslices = nil
+	m.Unlock()
+
+	m.dSliceWG.Wait()
 }
 
 func (m *baseMeta) expireTime() int64 {
@@ -2401,11 +2407,14 @@ func (m *baseMeta) deleteSlice(id uint64, size uint32) {
 	if id == 0 || m.conf.MaxDeletes == 0 {
 		return
 	}
-	if m.dslices != nil {
+	m.Lock()
+	dslices := m.dslices
+	m.Unlock()
+	if dslices != nil {
 		select {
 		case <-m.sessCtx.Done():
 			return
-		case m.dslices <- Slice{Id: id, Size: size}:
+		case dslices <- Slice{Id: id, Size: size}:
 		}
 	} else {
 		m.deleteSlice_(id, size)

